platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1127,18 +1477,115 @@ namespace platoon_strategic_ihp
+        // UCLA: conditions for cut-in join; platoon leader --> leading with operation
+        else if (isCutinJoin)
+        {
+            /* We are currently checking two basic JOIN conditions:
+             *     1. The size limitation on current platoon based on the plugin's parameters.
+             *     2. Calculate how far away the joining vehicle to the platoon (longitudinal & lateral),  
+             *        Determine if it was in a reasonable distance for cut-in join.
+             */ 
+            
+            cav_msgs::MobilityHeader msgHeader = msg.header;
+            std::string params = msg.strategy_params;
+            std::string applicantId = msgHeader.sender_id;
+            ROS_DEBUG_STREAM("Receive mobility JOIN request from " << applicantId << " and PlanId = " << msgHeader.plan_id);
+            ROS_DEBUG_STREAM("The strategy parameters are " << params);
+            if (params == "")
+            {
+                ROS_DEBUG_STREAM("The strategy parameters are empty, return no response");
+                return MobilityRequestResponse::NO_RESPONSE;
+            }
+            
+            // UCLA: For "CUT_IN_FROM_SIDE" message, the strategy params is : "SIZE:xx,SPEED:xx,DTD:xx" ---> NEED TO ADD IN HEADER FILE
+            std::vector<std::string> inputsParams;
+            boost::algorithm::split(inputsParams, params, boost::is_any_of(","));
+
+            std::vector<std::string> applicantSize_parsed;
+            boost::algorithm::split(applicantSize_parsed, inputsParams[0], boost::is_any_of(":"));
+            int applicantSize = std::stoi(applicantSize_parsed[1]);
+            ROS_DEBUG_STREAM("applicantSize: " << applicantSize);
+
+            std::vector<std::string> applicantCurrentSpeed_parsed;
+            boost::algorithm::split(applicantCurrentSpeed_parsed, inputsParams[1], boost::is_any_of(":"));
+            double applicantCurrentSpeed = std::stod(applicantCurrentSpeed_parsed[1]);
+            ROS_DEBUG_STREAM("applicantCurrentSpeed: " << applicantCurrentSpeed);
+
+            // Calculate downtrack (m) base on ecef. 
+            lanelet::BasicPoint2d incoming_pose = ecef_to_map_point(msg.location);
+            double applicantCurrentDtd = wm_->routeTrackPos(incoming_pose).downtrack;
+            ROS_DEBUG_STREAM("applicantCurrentmemberUpdates from ecef pose: " << applicantCurrentDtd);
+
+            // UCLA: Use crosstrack to determine condition to check if joining vehicle is close by
+            double applicant_crosstrack = wm_->routeTrackPos(incoming_pose).crosstrack;
+            ROS_DEBUG_STREAM("applicant_crosstrack from ecef pose: " << applicant_crosstrack);
+            double two_lane_cross_error = 2*config_.maxCrosstrackError + 3.5;
+
+            // cut-in vehicle and platoon are in adjacent lanes.
+            bool isLateralNearBy = (abs(applicant_crosstrack - current_crosstrack_) < two_lane_cross_error);
```

Comment:
Why not call isVehicleNextLane() for consistency and code reuse (better maintainability).  This is the third different logic used for such a test.